### PR TITLE
[libpsl] Fix libicu+undefined build

### PR DESCRIPTION
**Pull request by rockdaboot**
Libicu has problems with UBSAN that I can't easily solve (project/icu/project.yaml also excludes UBSAN).
This PR fixes the build by skipping libicu/UBSAN combination.
It also fixes iconv() detection in ./configure runs by disabling asan/ubsan leak check.

### CODE REVIEW

File: projects/libpsl/build.sh [hunk 1]
```diff
@@ -21,60 +21,81 @@ export CPPFLAGS="-I$DEPS_PATH/include"
...
 
 cd $SRC/icu/source
-./configure --disable-shared --enable-static --disable-extras --disable-icuio --disable-layoutex \
-  --disable-tests --disable-samples --with-data-packaging=static --prefix=$DEPS_PATH
+if test $SANITIZER = "undefined"; then
+  UBSAN_OPTIONS=detect_leaks=0 \
+  CFLAGS="$CFLAGS -fno-sanitize=function,vptr" \
+  CXXFLAGS="$CXXFLAGS -fno-sanitize=function,vptr" \
+  CPPFLAGS="$CPPFLAGS -fno-sanitize=function,vptr" \
+  LDFLAGS="$LDFLAGS -fsanitize=undefined" \
```
Comment (Dor1s) [inline projects/libpsl/build.sh]: It doesn't work without LDFLAGS, right?

File: projects/libpsl/build.sh [hunk 2]
```diff
@@ -21,60 +21,81 @@ export CPPFLAGS="-I$DEPS_PATH/include"
 export LDFLAGS="-L$DEPS_PATH/lib"
 
 cd $SRC/icu/source
-./configure --disable-shared --enable-static --disable-extras --disable-icuio --disable-layoutex \
-  --disable-tests --disable-samples --with-data-packaging=static --prefix=$DEPS_PATH
+if test $SANITIZER = "undefined"; then
```
Comment (Dor1s) [inline projects/libpsl/build.sh]: Let's remove this `if` statement. We should not rely on env variables other than described in https://github.com/google/oss-fuzz/blob/master/docs/new_project_guide.md#buildsh-script-environment

It's safe to always add `-fno-sanitize=` flags to CFLAGS and others even for non-UBSan configs.

**Comment by rockdaboot** [comment]
Thanks for review. The changes worked locally.

**Comment by Dor1s** [comment]
Thanks Tim!

**Pull request merged**